### ISSUE TITLE
Problem linking with uuid on OS X Yosemite

### ISSUE THREAD

**Issue by rcodddow** [ros/nodelet_core#26]
This seems to be the same issue as in bond_core https://github.com/ros/bond_core/issues/6

I have already tested the changes from https://github.com/ros/bond_core/pull/7 in nodelet which
seems to resolve the issue

```
2 warnings generated.
Linking CXX shared library /Users/robert/ros_catkin_ws/devel_isolated/nodelet/lib/libnodeletlib.dylib
[ 93%] Built target nodeletlib
Scanning dependencies of target nodelet
[100%] Building CXX object CMakeFiles/nodelet.dir/src/nodelet.cpp.o
Linking CXX executable /Users/robert/ros_catkin_ws/devel_isolated/nodelet/lib/nodelet/nodelet
ld: library not found for -luuid
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [/Users/robert/ros_catkin_ws/devel_isolated/nodelet/lib/nodelet/nodelet] Error 1
make[1]: *** [CMakeFiles/nodelet.dir/all] Error 2
make: *** [all] Error 2
<== Failed to process package 'nodelet': 
  Command '/Users/robert/ros_catkin_ws/install_isolated/env.sh make -j8 -l8' returned non-zero exit status 2

Reproduce this error by running:
==> cd /Users/robert/ros_catkin_ws/build_isolated/nodelet && /Users/robert/ros_catkin_ws/install_isolated/env.sh make -j8 -l8

Command failed, exiting.
```

**Comment by esteve** [comment]
Thanks for testing ros/bond_core#7 and for reporting this issue. Seeing that this may be a recurring issue with OSX Yosemite, I'm going to add a pull request to `cmake_modules` to add the `FindUUID.cmake` module, and update the dependencies accordingly.

**Comment by esteve** [comment]
This issue was addressed in 031e59ec7020dca932a40665d477cdc6f72c4622 and d7120c0db934501269abbe5fd7cdd206ab6ba5f9
